src/components/MenuItem.js

```diff
@@ -25,38 +29,46 @@ const propTypes = {
 
 const defaultProps = {
     shouldShowRightArrow: false,
+    wrapperStyle: null,
 };
 
 const MenuItem = ({
     onPress,
     icon,
     title,
     shouldShowRightArrow,
-}) => (
-    <Pressable
-        onPress={onPress}
-        style={({hovered}) => ([
-            styles.createMenuItem,
-            hovered && {backgroundColor: themeColors.buttonHoveredBG},
-        ])}
-    >
-        <View style={styles.flexRow}>
-            <View style={styles.createMenuIcon}>
-                <Icon src={icon} />
-            </View>
-            <View style={styles.justifyContentCenter}>
-                <Text style={[styles.createMenuText, styles.ml3]}>
-                    {title}
-                </Text>
-            </View>
-        </View>
-        {shouldShowRightArrow && (
-            <View style={styles.createMenuIcon}>
-                <Icon src={ArrowRight} />
+    wrapperStyle,
+}) => {
+    // Flatten the wrapperStyle in case it's an array of styles
+    const flatWrapperStyle = StyleSheet.flatten(wrapperStyle);
```

Comment:
I'm not sure if I agree. I don't view this as being less clear, and I think it's a bit awkward to do stuff like 
```
<MenuItem
    wrapperStyle={[singleStyle]}
    ...
```
especially considering that other style props don't require the use of arrays. To me, it makes more sense to use arrays when necessary, and otherwise pass single objects. 

Ultimately though, it's a small change I'm willing to make if you aren't convinced.

Reply:
I suppose the other direction we could go with this is to say `MenuItem` is only being called with the `wrapperStyle` prop in one place, and in that call, the prop is a single object. So we remove this pre-optimization line, and make it default to an empty object that we don't spread. Thoughts?

Reply:
>To me, it makes more sense to use arrays when necessary, and otherwise pass single objects.

I'm gonna guess you mean it's awkward to pass an array when it only has one item in it. I don't mind this personally, but the perspective makes sense to me (despite my initial suggestion to use an array). 

So in this case should we stick with using an object type for this prop?

My ultimate concern is that we pick a type and stick with it, give it a good name, and try to keep things consistent. It could be an array or object. But if the prop is called `wrapperStyle` it probably takes just one style object, if the prop is named `wrapperStyles` it probably takes an array of style objects, etc.

Reply:
> My ultimate concern is that we pick a type and stick with it, give it a good name, and try to keep things consistent. It could be an array or object. But if the prop is called wrapperStyle it probably takes just one style object, if the prop is named wrapperStyles it probably takes an array of style objects, etc.

NAB but I think my stance on this is that it makes sense for our custom style props to take both an array of objects or a single object, because that's how the style props work for other RN components like `View` or `TouchableOpacity`, etc...

Reply:
Cool to discuss outside of this PR, but I think we can maybe split this into two different kinds of cases and probably adopt some standard practice around it then fix it up everywhere. I'd maybe propose something like this...

- a component that accepts one or more specifically named style props that modify one or more of the child elements should indicate what type it expects

e.g. 

`outerViewStyle` - takes one style object
`innerViewStyle` - takes one style object

OR 

`outerViewStyles` - takes an array of style objects
`innerViewStyles` - takes an array of style objects

- A component that lightly wraps another component and only has one `style` prop may use `PropTypes.any` if the style it passed directly to a child `style` prop on some RN element.

e.g. 

```jsx
<SomeSpecialInput 
    style={} // can be [] or {} a.k.a. PropTypes.any but must pass those styles directly to a child style
/>
```

Something that really bugs me about React Native is the inconsistency with some components e.g. it's confusing to me how something can have a `style` prop, but also have a `contentContainerStyle` prop.

Reply:
Gonna create an issue for this.